### PR TITLE
Test strategy with jest

### PR DESCRIPTION
**Pull request by ChaituVR**
Tests few scenarios including those mentioned in https://github.com/snapshot-labs/snapshot.js/issues/212

Tests the following cases: (also prints scores and times in console)

image

If this is good, can try to add other scenarios as well, so that user creating strategy will know exactly what we need from a strategy before raising a PR

See in action: 
https://github.com/ChaituVR/snapshot.js/runs/2603281800?check_suite_focus=true

### CODE REVIEW

**Comment by ChaituVR** [comment]
@bonustrack  we can use this if we remove 500 addresses test cases? or what other test cases can we remove
image

File: test/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      // wait for all logs to be printed (bug: printed after results)
+      await new Promise((r) => setTimeout(r, 2000));
+    }, 20000);
+
+    it(`getScores ${moreArg || 500} should return an array`, () => {
+      expect(scoresMore).toBeTruthy();
+      expect(Array.isArray(scoresMore)).toBe(true);
+    });
+
+    it(`The strategy for ${moreArg || 500} addresses should take less than 10 seconds to resolve`, () => {
```
Comment (bonustrack) [inline test/index.spec.ts:104]: Can you use 15sec here?

File: test/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    example.addresses,
+    example.snapshot
+  );
+}
+
+describe(`Snapshot Strategy Test:\n\nStrategy: "${strategy}"\nQuery: "${example.name}"\n\n`, () => {
+  let scores = null;
+  let getScoresTime = null;
+
+  it('Should run startegy without any errors', async () => {
```
Comment (bonustrack) [inline test/index.spec.ts:50]:
```suggestion
  it('Should run strategy without any errors', async () => {
```

**Comment by ChaituVR** [comment]
@bonustrack fixed those suggestions

image

**Review by bonustrack — APPROVED** [review]

**Pull request merged**